codeDeploy/elb/common_functions.sh

```diff
@@ -25,18 +28,70 @@ export PATH="$PATH:/usr/bin:/usr/local/bin"
 # If true, all messages will be printed. If false, only fatal errors are printed.
 DEBUG=true
 
+# If true, all commands will have a initial jitter - use this if deploying to significant number of instances only
+INITIAL_JITTER=false
+
 # Number of times to check for a resouce to be in the desired state.
-# We have connection draining set to 300 seconds for production ELBs,
-# so if any traffic does get stuck for a while, this is going to cause a failed deployment.
-# Unless our wait attempts here is greater the the connection draining setting.
-WAITER_ATTEMPTS=360
+WAITER_ATTEMPTS=60
 
 # Number of seconds to wait between attempts for resource to be in a state.
-WAITER_INTERVAL=1
+WAITER_INTERVAL=3
 
 # AutoScaling Standby features at minimum require this version to work.
 MIN_CLI_VERSION='1.3.25'
 
+# Create a flagfile for each deployment
+FLAGFILE="/tmp/asg_codedeploy_flags-$DEPLOYMENT_GROUP_ID-$DEPLOYMENT_ID"
+
+# Handle ASG processes
+HANDLE_PROCS=true
```

Comment:
these are all default AWS scripts except for this setting which we've turned on to get all the ASG supporting goodness